src/test.rs

```diff
@@ -343,3 +344,20 @@ fn check_test_environment() {
     #[cfg(target_os = "linux")]
     fn test_iface() {}
 }
+
+#[test]
+fn test_get_network_interfaces_ips() {
+    test_iface_ips();
```

Comment:
So, I agree that this is a suboptimal way to test the method, but it seems that a prerequisite for more comprehensive and reliable tests would involve a fairly significant rewrite of almost all the code in `get_network_interfaces_impl`. 

I'd be happy to attempt such a rewrite if you thought it advisable/worth it, but as I mentioned my rust-fu is weak, so I'd probably need a fair bit of feedback on my attempts to do so. If that's the route you'd like to go, we could maybe just get rid of the test for this PR, and then tackle that in another PR.

Open to other suggestions, etc.